pkg/parser/type.go

```diff
@@ -175,3 +175,23 @@ func (t *Type) sameComposite(t2 *Type) bool {
 	}
 	return false
 }
+
+func combineTypes(types []*Type) *Type {
+	combinedT := types[0]
+	for _, t := range types[1:] {
+		if combinedT.accepts(t) {
+			continue
+		}
+		if t.accepts(combinedT) {
+			combinedT = t
+			continue
+		}
+		if t.sameComposite(combinedT) {
+			combinedT = &Type{Name: t.Name, Sub: ANY_TYPE}
+			continue
+		}
+		return ANY_TYPE
+	}
+	return combinedT
+}
+
```

Comment:
nit nit: extra blank line at EOF. It does get removed in the next commit though, so not really a worry